components/core/src/DictionaryWriter.hpp

```diff
@@ -110,12 +100,7 @@ class DictionaryWriter {
 
 template <typename DictionaryIdType, typename EntryType>
 DictionaryWriter<DictionaryIdType, EntryType>::~DictionaryWriter () {
-    if (false == m_uncommitted_entries.empty()) {
-        SPDLOG_ERROR("DictionaryWriter contains uncommitted entries while being destroyed - possible data loss.");
-    }
-    for (const auto& value_entry_pair : m_value_to_entry) {
-        delete value_entry_pair.second;
-    }
+    // Do nothing
```

Comment:
If the destructor is empty, you can delete it entirely or replace it with `= default` in the class declaration.